Examples/Tests/reduced_diagnostics_python/PICMI_inputs_reduced.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python3
+import numpy as np
+from pywarpx import picmi
+
+# Physical constants
+c = picmi.constants.c
+
+max_steps = 200
+
+# Create grid
+grid = picmi.Cartesian3DGrid(
+    number_of_cells=[32, 32, 32],
+    lower_bound=[-1., -1., -1.],
+    upper_bound=[1., 1., 1.],
+    lower_boundary_conditions=['periodic', 'periodic','periodic'],
+    upper_boundary_conditions=['periodic', 'periodic','periodic'],
+    lower_boundary_conditions_particles=['periodic', 'periodic','periodic'],
+    upper_boundary_conditions_particles=['periodic', 'periodic','periodic'],
+    warpx_max_grid_size=32)
+
+electrons = picmi.Species(
+        particle_type='electron',
+        name=f'electrons',
+        initial_distribution= picmi.UniformDistribution(
+                                density=1.e14,
+                                rms_velocity = [0.035*c]*3
+        ))
+photons = picmi.Species(
+        particle_type='photon',
+        name='photons',
+        initial_distribution=picmi.UniformDistribution(
+                                density=1.e14,
+                                rms_velocity = [0.2*c]*3)
+)
+
+protons = picmi.Species(
+        particle_type='proton',
+        name=f'protons',
+        initial_distribution=picmi.UniformDistribution(
+                                density=1.e14
+        ))
+
+species_list = [electrons, protons, photons]
+
+
+# reduced diagnostics
+EP = picmi.ReducedDiagnostic(name='EP', type='ParticleEnergy',intervals=200)
+NP = picmi.ReducedDiagnostic(name='NP', type='ParticleNumber',intervals=200)
+EF = picmi.ReducedDiagnostic(name='EF', type='FieldEnergy',intervals=200)
+PP = picmi.ReducedDiagnostic(name='PP', type='ParticleMomentum',intervals=200)
+PF = picmi.ReducedDiagnostic(name='PF', type='FieldMomentum',intervals=200)
+MF = picmi.ReducedDiagnostic(name='MF', type='FieldMaximum',intervals=200)
+MR = picmi.ReducedDiagnostic(name='MR', type='RhoMaximum',intervals=200)
+FP = picmi.ReducedDiagnostic(name='FP',
+                                type='FieldProbe',
+                                intervals=200,
+                                #warpx_probe_geometry='Point',
+                                warpx_x_probe = 0.53125,
```

Comment:
The `warpx` prefix is not needed for these arguments since they are essential for the description (and wouldn't be specific to WarpX).

Reply:
On a related note, the reduced diagnostics aren't in the PICMI standard.  In a sense, this whole PR is maybe breaking with the spirit of PICMI and is something I wanted to ask about

Reply:
Re: your comment, I was borrowing the `handle_init` function, which requires the warpx prefix.  That said, it is awkward and worth removing.

Reply:
actually, this doesn't have a straighforward simplification.  

-  to keep all reduced diagnostics in one PICMI class, some diagnostic specific inputs are passed as keyword arguments
- all keyword arguments have to be prefixed with the code prefix or errors are thrown

- one alternative is to have separate diagnostic classes for more complicated reduced diagnostics
- I chose not to pursue this direction, but it could be revisited if we think it is more annoying to prefix keyword arguments with "warpx_" than to have multiple reduced diagnostics classes

Reply:
Regarding the reduced diagnostics not being in the PICMI standard, that is OK for now. The interface can be worked out here and then later moved to PICMI.

Having separate classes for the more complicated diagnostics might be a good idea. This would make it easier to do error checking on the input arguments. Each diagnostic type that has different input parameters would get its own class. This would fit the pattern of the PICMI standard better, with all of the arguments explicitly defined in the class constructor.

Reply:
Thinking about this again, it may be a bit more work now, but would save work in the long term if this was added to the PICMI standard now (instead of only in WarpX). The classes would only have to be written once and the extra stuff needed to define the classes in WarpX would not be needed. This PR would be put on WIP status until that is ready and then this PR would update the required version of the picmi standard.